diff -ruN sdl2-2.28.4/Makefile.amigaos4 sdl2-2.28.4-patched/Makefile.amigaos4
--- sdl2-2.28.4/Makefile.amigaos4	2023-10-04 18:20:11.000000000 +0100
+++ sdl2-2.28.4-patched/Makefile.amigaos4	2024-01-28 22:00:07.328322572 +0000
@@ -5,11 +5,12 @@
 CC = ppc-amigaos-gcc
 CXX = ppc-amigaos-g++
 STRIP = ppc-amigaos-strip
+CRT ?= newlib
 CATCOMP = catcomp
 
 AMIGADATE = $(shell date +"%-d.%m.%Y")
 
-CFLAGS  = -gstabs -O2 -Wall -fPIC -fcommon -I./include -D__AMIGADATE__=\"$(AMIGADATE)\"
+CFLAGS  = -mcrt=$(CRT) -DSDL_GL4ES -gstabs -O2 -Wall -fPIC -fcommon -I./include -D__AMIGADATE__=\"$(AMIGADATE)\"
 
 TARGET_STATIC  = libSDL2.a
 TARGET_SHARED  = libSDL2-2.28.so
@@ -79,7 +80,7 @@
 	$(RANLIB) $@
 
 $(TARGET_SHARED): $(OBJECTS) $(VERSION_OBJECT)
-	$(CC) -shared -Wl,-soname,$(TARGET_SHARED) -o $(TARGET_SHARED) $(OBJECTS) $(VERSION_OBJECT)
+	$(CC) -mcrt=$(CRT) -shared -Wl,-soname,$(TARGET_SHARED) -o $(TARGET_SHARED) $(OBJECTS) $(VERSION_OBJECT)
 
 $(PREFS_PROGRAM): $(PREFS_OBJECT)
 	$(CC) -o $@ $^ # -lauto
diff -ruN sdl2-2.28.4/src/video/amigaos4/SDL_os4opengl.c sdl2-2.28.4-patched/src/video/amigaos4/SDL_os4opengl.c
--- sdl2-2.28.4/src/video/amigaos4/SDL_os4opengl.c	2023-10-04 18:20:11.000000000 +0100
+++ sdl2-2.28.4-patched/src/video/amigaos4/SDL_os4opengl.c	2024-01-28 21:57:56.577398692 +0000
@@ -20,7 +20,7 @@
 */
 #include "../../SDL_internal.h"
 
-#if SDL_VIDEO_DRIVER_AMIGAOS4
+#if defined (SDL_VIDEO_OPENGL) && !defined (SDL_GL4ES)
 
 #include <proto/graphics.h>
 #include <proto/minigl.h>
@@ -266,7 +266,7 @@
     } else {
         OS4_GL_LogLibraryError();
     }
-    
+
     return result;
 }
 
diff -ruN sdl2-2.28.4/src/video/amigaos4/SDL_os4openglwrapper.c sdl2-2.28.4-patched/src/video/amigaos4/SDL_os4openglwrapper.c
--- sdl2-2.28.4/src/video/amigaos4/SDL_os4openglwrapper.c	2023-10-04 18:20:11.000000000 +0100
+++ sdl2-2.28.4-patched/src/video/amigaos4/SDL_os4openglwrapper.c	2024-01-28 22:27:59.920348691 +0000
@@ -23,7 +23,7 @@
 
 /* wrapper functions for MiniGL */
 
-#if SDL_VIDEO_DRIVER_AMIGAOS4
+#if defined (SDL_VIDEO_OPENGL) && !defined (SDL_GL4ES)
 
 #include <GL/gl.h>
 #include <GL/glu.h>
diff -ruN sdl2-2.28.4/src/video/amigaos4/SDL_os4video.c sdl2-2.28.4-patched/src/video/amigaos4/SDL_os4video.c
--- sdl2-2.28.4/src/video/amigaos4/SDL_os4video.c	2023-10-04 18:20:11.000000000 +0100
+++ sdl2-2.28.4-patched/src/video/amigaos4/SDL_os4video.c	2024-01-28 21:57:56.580732050 +0000
@@ -375,10 +375,15 @@
         _this->gl_config.major_version,
         _this->gl_config.minor_version);
 
+#ifdef SDL_GL4ES
+        OS4_SetMiniGLFunctions(_this);
+        return OS4_GL_LoadLibrary(_this, path);
+#else
     if (OS4_IsMiniGL(_this)) {
         OS4_SetMiniGLFunctions(_this);
         return OS4_GL_LoadLibrary(_this, path);
     }
+#endif
 
 #if SDL_VIDEO_OPENGL_ES2
     if (OS4_IsOpenGLES2(_this)) {
